packages/react-testing/src/types.ts

```diff
@@ -112,5 +112,6 @@ export interface Node<Props> {
   ): MaybeFunctionReturnType<NonNullable<Props[K]>>;
   triggerKeypath<T = unknown>(keypath: string, ...args: unknown[]): T;
 
+  debug(maxDepth?: number): string;
```

Comment:
could also be handy to have an `omitChildren` flag.

Reply:
also `maxDepth` is a poor choice for a name, it really represents prop object expansion depth. perhaps a better approach is `{propsDepth?: number; childDepth?: number}`?

Reply:
I'm not sure when I'd want those two to be different values, or if i did that I'd remember that as an option. A single value for both feels more intuitive.

Reply:
i can see some legit use cases.
> I want to see max detail for all props on a single component with no children
`wrapper.findWhere(({type}) => type && type.name === 'Header').debug({verbosity: 5, depth: 0})`
```tsx
<Header
  breadcrumbs={[{content: 'Settings', url: '/admin/settings'}]}
  title="Capital"
  separator
  polaris={{
    appBridge: undefined,
    intl: {
      translate: ['Function anonymous'],
      translation: {
        Polaris: {
          ActionMenu: {RollupActions: {rollupButton: 'Actions'}},
          Autocomplete: {spinnerAccessibilityLabel: 'Loading'},
          Avatar: {
            label: 'Avatar',
            labelWithInitials: 'Avatar with initials {initials}',
          },
          Badge: {
            PROGRESS_LABELS: {
              complete: 'Complete',
              incomplete: 'Incomplete',
              partiallyComplete: 'Partially complete',
            },
            STATUS_LABELS: {
              attention: 'Attention',
              info: 'Info',
              new: 'New',
              success: 'Success',
              warning: 'Warning',
            },
          },
          Button: {spinnerAccessibilityLabel: 'Loading'},
          Common: {
            cancel: 'Cancel',
            checkbox: 'checkbox',
            clear: 'Clear',
            close: 'Close',
            more: 'More',
            newWindowAccessibilityHint: '(opens a new window)',
            submit: 'Submit',
            undo: 'Undo',
          },
          ContextualSaveBar: {discard: 'Discard', save: 'Save'},
          DataTable: {
            navAccessibilityLabel:
              'Scroll table {direction} one column',
            sortAccessibilityLabel: 'sort {direction} by',
            totalsRowHeading: 'Totals',
          },
          DatePicker: {
            daysAbbreviated: {
              friday: 'Fr',
              monday: 'Mo',
              saturday: 'Sa',
              sunday: 'Su',
              thursday: 'Th',
              tuesday: 'Tu',
              wednesday: 'We',
            },
            months: {
              april: 'April',
              august: 'August',
              december: 'December',
              february: 'February',
              january: 'January',
              july: 'July',
              june: 'June',
              march: 'March',
              may: 'May',
              november: 'November',
              october: 'October',
              september: 'September',
            },
            nextMonth: 'Show next month, {nextMonth} {nextYear}',
            previousMonth:
              'Show previous month, {previousMonthName} {showPreviousYear}',
            today: 'Today ',
          },
          DiscardConfirmationModal: {
            message:
              'If you discard changes, you’ll delete any edits you made since you last saved.',
            primaryAction: 'Discard changes',
            secondaryAction: 'Continue editing',
            title: 'Discard all unsaved changes',
          },
          DropZone: {
            FileUpload: {
              actionHintFile: 'or drop files to upload',
              actionHintImage: 'or drop images to upload',
              actionTitleFile: 'Add file',
              actionTitleImage: 'Add image',
              label: 'Upload file',
            },
            errorOverlayTextFile: 'File type is not valid',
            errorOverlayTextImage: 'Image type is not valid',
            overlayTextFile: 'Drop file to upload',
            overlayTextImage: 'Drop image to upload',
          },
          EmptySearchResult: {altText: 'Empty search results'},
          Filters: {
            cancel: 'Cancel',
            clear: 'Clear',
            clearAllFilters: 'Clear all filters',
            clearLabel: 'Clear {filterName}',
            done: 'Done',
            filter: 'Filter {resourceName}',
            moreFilters: 'More filters',
            noFiltersApplied: 'No filters applied',
          },
          Frame: {
            Navigation: {
              closeMobileNavigationLabel: 'Close navigation',
            },
            skipToContent: 'Skip to content',
          },
          Icon: {
            backdropWarning:
              'The {color} icon doesn’t accept backdrops. The icon colors that have backdrops are: {colorsWithBackDrops}',
          },
          Modal: {
            iFrameTitle: 'body markup',
            modalWarning:
              'These required properties are missing from Modal: {missingProps}',
          },
          Pagination: {
            next: 'Next',
            pagination: 'Pagination',
            previous: 'Previous',
          },
          ProgressBar: {
            exceedWarningMessage:
              'Values passed to the progress prop shouldn’t exceed 100. Setting {progress} to 100.',
            negativeWarningMessage:
              'Values passed to the progress prop shouldn’t be negative. Resetting {progress} to 0.',
          },
          ResourceList: {
            BulkActions: {
              actionsActivatorLabel: 'Actions',
              moreActionsActivatorLabel: 'More actions',
              warningMessage:
                'To provide a better user experience. There should only be a maximum of {maxPromotedActions} promoted actions.',
            },
            DateSelector: {
              FilterLabelForValue: {
                coming_month: 'next month',
                coming_quarter: 'in the next 3 months',
                coming_week: 'next week',
                coming_year: 'in the next year',
                on_or_after: 'after {date}',
                on_or_before: 'before {date}',
                past_month: 'in the last month',
                past_quarter: 'in the last 3 months',
                past_week: 'in the last week',
                past_year: 'in the last year',
              },
              SelectOptions: {
                ComingMonth: 'next month',
                ComingQuarter: 'in the next 3 months',
                ComingWeek: 'next week',
                ComingYear: 'in the next year',
                OnOrAfter: 'on or after',
                OnOrBefore: 'on or before',
                PastMonth: 'in the last month',
                PastQuarter: 'in the last 3 months',
                PastWeek: 'in the last week',
                PastYear: 'in the last year',
              },
              dateFilterLabel: 'Select a value',
              dateValueError: 'Match YYYY-MM-DD format',
              dateValueLabel: 'Date',
              dateValuePlaceholder: 'YYYY-MM-DD',
            },
            FilterControl: {
              textFieldLabel: 'Search {resourceNamePlural}',
            },
            FilterCreator: {
              addFilterButtonLabel: 'Add filter',
              filterButtonLabel: 'Filter',
              selectFilterKeyPlaceholder: 'Select a filter…',
              showAllWhere: 'Show all {resourceNamePlural} where:',
            },
            FilterValueSelector: {
              selectFilterValuePlaceholder: 'Select a filter…',
            },
            Item: {
              actionsDropdown: 'Actions dropdown',
              actionsDropdownLabel:
                'Actions for {accessibilityLabel}',
              viewItem: 'View details for {itemName}',
            },
            a11yCheckboxDeselectAllMultiple:
              'Deselect all {itemsLength} {resourceNamePlural}',
            a11yCheckboxDeselectAllSingle:
              'Deselect {resourceNameSingular}',
            a11yCheckboxSelectAllMultiple:
              'Select all {itemsLength} {resourceNamePlural}',
            a11yCheckboxSelectAllSingle:
              'Select {resourceNameSingular}',
            allItemsSelected:
              'All {itemsLength}+ {resourceNamePlural} in your store are selected.',
            ariaLivePlural: '{itemsLength} items',
            ariaLiveSingular: '{itemsLength} item',
            defaultItemPlural: 'items',
            defaultItemSingular: 'item',
            emptySearchResultDescription:
              'Try changing the filters or search term',
            emptySearchResultTitle: 'No {resourceNamePlural} found',
            loading: 'Loading {resource}',
            selectAllItems:
              'Select all {itemsLength}+ {resourceNamePlural} in your store',
            selectButtonText: 'Select',
            selected: '{selectedItemsCount} selected',
            showing: 'Showing {itemsCount} {resource}',
            sortingLabel: 'Sort by',
          },
          SkeletonPage: {loadingLabel: 'Page loading'},
          Spinner: {
            warningMessage:
              'The color {color} is not meant to be used on {size} spinners. The colors available on large spinners are: {colors}',
          },
          Tabs: {toggleTabsLabel: 'More tabs'},
          Tag: {ariaLabel: 'Remove {children}'},
          TextField: {
            characterCount: '{count} characters',
            characterCountWithMaxLength:
              '{count} of {limit} characters used',
          },
          TopBar: {
            SearchField: {
              clearButtonLabel: 'Clear',
              search: 'Search',
            },
            toggleMenuLabel: 'Toggle menu',
          },
        },
      },
    },
    link: undefined,
    scrollLockManager: {locked: false, scrollLocks: 0},
    stickyManager: {
      handleResize: ['Function debounced'],
      handleScroll: ['Function debounced'],
      stickyItems: [],
      stuckItems: [],
      topBarOffset: 0,
    },
    theme: {logo: null},
  }}
>
  {/* <1 child... /> */}
</Header>
```

Reply:
Ah that makes more sense, thanks! I think i'd go for a 2 args approach with some sensible-ish rather than an object. Seeing children `.debug(childrenMaxDepth = 5, propsMaxDepth = 1)`.


It's at this point that I mutter something something Polaris being on a warpath to kill our WithAppProvider component (the only HoC remaining in v4), and [we're down to <15 usages](https://github.com/Shopify/polaris-react/issues/1995), but that won't help the reset of web being very HoC happy.